specification/validated-relationships-service-api.yaml

```diff
@@ -1107,119 +1209,97 @@ components:
-              items:
-                type: object
-                properties:
-                  system:
-                    type: string
-                    description: Codesystem URL for the proxy's NHS number.
-                    enum: ["https://fhir.nhs.uk/Id/nhs-number"]
-                  value:
-                    type: string
-                    description: The proxy's NHS number.
-                    example: "9000000017"
-            patient:
-              type: object
-              description: A reference to a patient the proxy is related to.
-              properties:
-                type:
-                  type: string
-                  description: FHIR resource type.
-                  enum: ["Patient"]
-                identifier:
-                  type: object
-                  description: The patient's NHS number. The primary identifier of a patient, unique within NHS England and Wales. Always 10 digits and must be a [valid NHS number](https://www.datadictionary.nhs.uk/attributes/nhs_number.html).
-                  properties:
-                    system:
-                      type: string
-                      description: Codesystem URL for the patient's NHS number.
-                      enum: ["https://fhir.nhs.uk/Id/nhs-number"]
-                    value:
-                      type: string
-                      description: The patient's NHS number.
-                      example: "9000000009"
-            relationship:
-              type: array
-              description: How the proxy is related to the patient.
-              items:
-                type: object
-                properties:
-                  coding:
-                    type: array
-                    description: FHIR coding array.
-                    items:
-                      type: object
-                      properties:
-                        system:
-                          type: string
-                          description: FHIR codesystem.
-                          default: "http://terminology.hl7.org/CodeSystem/v3-RoleCode"
-                          enum:
-                            [
-                              "http://terminology.hl7.org/CodeSystem/v3-RoleCode",
-                              "https://fhir.hl7.org.uk/CodeSystem/UKCore-AdditionalPersonRelationshipRole"
-                            ]
-                        code:
-                          type: string
-                          description: FHIR relationship type code.
-                          enum:
-                            - "MTH"
-                        display:
-                          type: string
-                          description: FHIR relationship type.
-                          enum:
-                            - "mother"
+              enum:
+                - match
+                - include
+                - outcome
+              example: match
```

Comment:
So what does "Mode" actually do - and how is it implemented?
"match" is fairly obvious - but does this enforce patient Ids to be passed in - does this fail validation if not?
"include" does this mean it fails if it *doesn't* include the passed in Id?  Or what if it is one of - do we return them all?
"Outcome" means just return all results?

Reply:
So it doesn't DO anything per say, it is an indicator as to why the entry is being returned in the bundle (so it's not instructive, it's informational in the response and we just need to set the value for each bundle entry to the appropriate value based on the below:

"match" - means it's the core resource that was being searched for, so in our case if it's the RelatedPerson endpoint 9for example), that's the entry that will have a search.mode ="match"

"include" - this indicates that the resource entry is in the bundle because the "_include" query parameter has been provided in the request, and this resource is being included as a result i.e. when we specify `_include=RelatedPerson:patient` each patient entry we get back in the bundle should have search.mode = "include"

Just re-reviewed the bundle docs and it looks like `outcome` has now been deprecated, so i've updated the spec to remove that from the permissible values